envoy/http/async_client.h

```diff
@@ -219,6 +225,43 @@ class AsyncClient {
       return *this;
     }
 
+    StreamOptions& setRetryPolicy(const envoy::config::core::v3::RetryPolicy& p,
```

Comment:
better add comment about the difference from the above `setRetryPolicy(const envoy::config::route::v3::RetryPolicy& p)`?